### ISSUE TITLE
Support for SimpleSAMLphp 2.x?

### ISSUE THREAD

**Issue by MultcoJeremy** [backdrop-contrib/simplesamlphp_auth#36]
With the release of SimpleSAMLphp 2.x, the 1.x branch will hit EOL at the end of 2023.  Will this backdrop compatiable module be upgrade to work with SimpleSAMLphp 2.x?

**Comment by laryn** [comment]
@MultcoJeremy Yes, we’ll need to. Have you done any testing with this module and 2.x yet?

**Comment by MultcoJeremy** [comment]
We have D7 site that uses simplesamlphp_auth and just evaluating possible path forwards.  

FYI, the watchdog log error we are running into with D7 + SimpleSAMLphp_auth using SimpleSAMLphp 2.0.6 is...

`Error: Class 'SimpleSAML_Configuration' not found in _simplesaml_auth_autoload() (line 150 of /app/docroot/sites/all/modules/contrib/simplesamlphp_auth/simplesamlphp_auth.module).`

Actually we get a WSOD when trying SSP 2.0.6, it's only after running `drush dis simplesamlphp_auth` that you can login to D7 as user 1.

**Comment by MultcoJeremy** [comment]
What I'm guessing is going on is with SimpleSAMLphp (SSP) 2.x, the paths have changed, including you have to login as admin to even see what version of SSP is running.  Thus SimpleSAMLphp_auth is getting a null/invalid value for the version number and that somehow results in WSOD.

**Comment by laryn** [comment]
@MultcoJeremy Got it -- if you have any questions as you evaluate Backdrop, feel free to ping me. The Zulip chat is a good spot to get feedback. I think you'll like Backdrop. https://backdrop.zulipchat.com/

As far as your particular error, it seems likely that the 2.x branch removes some deprecated functionality which is causing that error. In fact, it sounds very similar to this deprecation notice that I fixed here in May: https://github.com/backdrop-contrib/simplesamlphp_auth/issues/19

**Comment by MultcoJeremy** [comment]
#19 backported to the D7 simplesamlphp_auth module did the trick for us.

**Comment by laryn** [comment]
@MultcoJeremy That's good news! Thanks for reporting back.

**Comment by laryn** [comment]
Closing this issue as we have confirmation that 2.x works fine with this module.
